### ISSUE TITLE
Make it easier to test classes that process RecognizedForm

### ISSUE THREAD

**Issue by yowl** [Azure/azure-sdk-for-net#20282]
**Library or service name.**
Cognitive Services FormsRecognizer
 

If you create a class to process a `RecognizedForm` it seems hard to test that class because you can't supply the inputs . The constructor requires objects, e.g. `DocumentResult` that can't be created.  So for example

```
        [Test]
        public void TestExtraction()
        {
            var it = new SomeClassThatProcessesARecognizedForm();
            var rf = new RecognizedForm(/* cant :-( */);

            it.GetSomeValue(rf).Should().Be(1);
        }
```

**Comment by jsquire** [comment]
Hi @yowl.  Thank you for your feedback and we regret that you're experiencing difficulties.  Assuming that you're using the `Azure.AI.FormRecognizer` library, the `FormRecognizerModelFactory` is a static class that can create instances for testing.  The [`RecognizedForm`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerModelFactory.cs#L408) is among the types that can be created using it.

**Comment by yowl** [comment]
@jsquire Thanks, that looks useful.  One extra that might be considered:  add the ability to create from json.  That way the workflow could be:

1. Create the model in fott-preview.azurewebsites.net
2. Run the model on an input
3. Save the json and use as input to the test
4. Develop and test the processor for the RecognizedForm from an actual result.  

Advantage being that you are sure you've got the construction of the RecognizedForm the same as returned from the analysis.

**Comment by maririos** [comment]
Thank you @yowl for bringing this up! We had an issue similar to this, but haven't heard from customers needing it. I linked it here and bumped the priority as I like your proposal of adding it to the `ModelFactory` class.
Also, FYI, currently the `FormRecognizerModelFactory` is missing all new types that have been added during the work on `3.1.0-beta`. This issue tracks that work https://github.com/Azure/azure-sdk-for-net/issues/16519

BTW, feel free to contribute to our SDK by opening PRs in the repo :)
